src/org/labkey/test/BaseWebDriverTest.java

```diff
@@ -1719,42 +1719,22 @@ public DataRegionTable getCohortDataRegionTable()
      */
     public void verifyCohortStatus(DataRegionTable table, String cohort, boolean enrolled)
     {
-        int row = getCohortRow(table, cohort);
+        int row = table.getRowIndex("Label", cohort);
         assertEquals("Enrollment state for cohort " + cohort, String.valueOf(enrolled).toLowerCase(), table.getDataAsText(row, "Enrolled").toLowerCase());
     }
 
-    /**
-     * Used by CohortTest and StudyCohortExportTest
-     * Retrieves the row for the cohort matching the label passed in
-     */
-    public int getCohortRow(DataRegionTable cohortTable, String cohort)
-    {
-        int row;
-        for (row = 0; row < cohortTable.getDataRowCount(); row++)
-        {
-            String s = cohortTable.getDataAsText(row, "Label");
-            if (0 == s.compareToIgnoreCase(cohort))
-            {
-                break;
-            }
-        }
-        return row;
-    }
-
     /**
      * Used by CohortTest and StudyCohortExportTest
      * Changes the enrolled status of the passed in cohort name
      */
     public void changeCohortStatus(DataRegionTable cohortTable, String cohort, boolean enroll)
     {
         // if the row does not exist then most likely the cohort passed in is incorrect
-        int rowIndex = getCohortRow(cohortTable, cohort);
+        int rowIndex = cohortTable.getRowIndex("Label", cohort);
         cohortTable.clickEditRow(rowIndex);
+        WebElement checkboxElement = Locator.name("quf_enrolled").waitForElement(getDriver(), 2000);
+        new org.labkey.test.components.html.Checkbox(checkboxElement).set(enroll);   // the ext4 checkbox doesn't correctly parse checked state of this html checkbox
```

Comment:
This is good cleanup. The Ext4 mention just seems like a non-sequitur.
```suggestion
        new org.labkey.test.components.html.Checkbox(checkboxElement).set(enroll);
```

Reply:
OK, so it turns out that using DataRegionTable.updateRow(index, map, validate) also does this